src/audio/module_adapter/module_adapter.c

```diff
@@ -245,6 +245,20 @@ int module_adapter_prepare(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
+	/* Check if simple copy client has only one source and one sink buffer to use a
+	 * simplified copy function. The multi_source_or_sink is set later to true
+	 * if module_adapter_bind() indicates multiple sources.
+	 */
+	if (mod->simple_copy && mod->num_input_buffers == 1 && mod->num_output_buffers == 1) {
+		mod->source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+		mod->sink = sink;
+		mod->multi_source_or_sink = false;
+	} else {
+		mod->source = NULL;
+		mod->sink = NULL;
+		mod->multi_source_or_sink = true;
+	}
+
```

Comment:
which part called first? prepare or bind? seems above code is duplicated with bind code.

Reply:
I don't think there's guaranteed order for those events. Do you see risk or ways to mitigate it?